premis_event_service/presentation.py

```diff
@@ -39,18 +47,24 @@ def premisEventXMLToObject(eventXML):
         doSimpleXMLAssignment(newEventObject, fieldName, eventXML, chain)
     linkingObjectIDNodes = getNodesByName(eventXML, "linkingObjectIdentifier")
     try:
-        Event.objects.get(event_identifier=identifier)
-        resp = HttpResponse('The uuid already exists')
-        resp.status_code = 409
-        return resp
-    except Exception, e:
+        Event.objects.get(event_identifier=newEventObject.event_identifier)
+        raise DuplicateEventError(newEventObject.event_identifier)
+    except Event.DoesNotExist as e:
         pass
+    except DuplicateEventError as e:
+        raise e
     # if we dont have a uuid looking event_identifier
     # create our own before we save.
     try:
         uuid.UUID(newEventObject.event_identifier)
     except Exception, e:
         newEventObject.event_identifier = uuid.uuid4().hex
+    if isinstance(newEventObject.event_date_time, datetime):
+        pass
+    else:
+        newEventObject.event_date_time = xsDateTime_parse(
+            newEventObject.event_date_time
+        )
```

Comment:
Is there a purpose to having the `if` block that only passes followed by an `else`, instead of an `if not isinstance(...`?